src/components/SearchBox.vue

```diff
@@ -2,7 +2,27 @@
   <form role="search" action="" @submit.prevent="onFormSubmit">
     <slot>
       <ais-input :search-store="searchStore" :placeholder="placeholder" :autofocus="autofocus"></ais-input>
-      <button type="submit" :class="bem('submit')">
+      <div v-if="showLoadingIndicator" :style="searchStore.isSearchStalled ? 'display:block;' : 'display:none;'" :class="bem('loading-indicator')" >
```

Comment:
I would probably go for `v-show="showLoadingIndicator && searchStore.isSearchStalled"` and remove the `style` attribute.

Reply:
Agreed for that it would make more sense given the vue-specifics. However I chose that in order to have consistency accross all implementations. This is the exact same markup and changes on it depending on the value of `isSearchStalled` in all implementations.